ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -340,6 +340,28 @@ def test_human_output(self):
         assert returncode == 0
         assert output == "No issues found."
 
+    def test_ipa_healthcheck_fips_enabled(self):
+        """
+        Test if FIPS is enabled and the check exists.
+
+        https://pagure.io/freeipa/issue/8951
+        """
+        returncode, output = run_healthcheck(self.master, output_type="json",
+                                             failures_only=False)
+        assert returncode == 0
+
+        cmd = self.master.run_command(['fips-mode-setup', '--is-enabled'],
+                                      raiseonerr=False)
+        returncode = cmd.returncode
+        for check in output:
+            if check["kw"].get("fips", None):
+                if check["kw"]["fips"] == "disabled":
+                    assert returncode == 2
+                elif check["kw"]["fips"] == "enabled":
+                    assert returncode == 0
+                else:
+                    assert returncode == 1
```

Comment:
Rather than only executing when fips is a keyword, I'd look for the specific check that should contain this. Otherwise if for some reason ipa-healthcheck is broken and not returning MetaCheck then this test will not catch it.

I think it's worthwhile to also run this with FIPS enabled. There are "fake" enable/disable methods for fips in the TestIntegration class that may be sufficient. They are currently unused AFAICT so no guarantees they will do the right thing.

Reply:
@rcritten the test is running in true FIPS env[1] with both FIPS enabled and DISABLED, however, I concur it should run with both in one pass.

[1] introduced with https://github.com/freeipa/freeipa-pr-ci/pull/452

Reply:
Sorry I looked but missed that in the test definition. Testing both in the temp commit would be great, just reference the passing tests for both.